presto-main/src/main/java/com/facebook/presto/sql/ExpressionUtils.java

```diff
@@ -104,7 +103,17 @@ public static Expression binaryExpression(LogicalBinaryExpression.Type type, Col
     {
```

Comment:
https://github.com/prestodb/presto/pull/10322/commits/49bbaf1aa5e9b6bb35c196cd3da3095ec0ee0155#diff-36b1f59a738ee41b16d448c90d39d890L160 seem redundant to it as this one produces better predicates (more balanced).

However, there are some additional optimizations there https://github.com/prestodb/presto/pull/10322/commits/49bbaf1aa5e9b6bb35c196cd3da3095ec0ee0155#diff-36b1f59a738ee41b16d448c90d39d890R197.

In the future we could merge those two ways of creating logical expressions.

Reply:
discussed offline, left for the future.